### PR TITLE
Render Single Component

### LINKED ISSUE #74: Support rendering manifests for a single component
To simplify developing components it should be possible to have Commodore render just the manifests generated by that component, ideally without requiring any external APIs. This will also be a useful building block for implementing automated validation and testing of components.

Potentially the changes required for #71 can be reused to implement this feature.

### Task deliverables

- [x] Manifests generated by a single component can be rendered by Commodore
- [x] No pre-setup workspace or external API interaction is required to render a component's manifests

### PR DESCRIPTION
**Pull request by srueg**
Render a single component by faking the required parts of the inventory. Additional classes can be specified with the `--values/-f` flag (multiple possible).
For the compilation to succeed, a user is responsible to provide all required parameters (which are not covered in the `defaults.yml`).

This changes the CLI structure: There's two new command groups: `catalog` and `component`. The previous `compile` and `clean` commands are moved to the `catalog` group and the `new-component` was moved together with the `compile-component` command into the `component` group:

```commodore component new```
```commodore component compile```
```commodore catalog compile```

~~This first implementation doesn't support secret refs. This means all used secret refs in the `defaults.yml` must be overridden with concrete values in the provided classes.~~
Vault secret refs are supported and get revealed to the path they reference (e.g. `?{vaultkv:${customer:name}/${cluster:name}/backup-cluster-objects/password}` becomes `t-silent-test-1234/c-green-test-1234/backup-cluster-objects/password`)

Closes #74

### CODE REVIEW

**Comment by srueg** [comment]
Not done yet / Caveats:
* The `argocd.libjsonnet` library is being faked with an empty object, therefore the Argo CD `application` manifest won't be rendered
* No jsonnet lib dependencies are created, therefore the `kube.libjsonnet` is missing (which breaks many components)

**Comment by srueg** [comment]
I fixed the remaining open points and was even able to find a solution to support vault references: By creating a fake VaultBackend, references are "revealed" to the path in Vault they specify.

File: commodore/component/compile.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
+import os
+from pathlib import Path as P
+import shutil
+import tempfile
+
+import click
+
+from commodore.config import Config
+from commodore.dependency_mgmt import fetch_jsonnet_libs, _relsymlink
```
Comment (simu) [inline commodore/component/compile.py:9]: Consider refactoring `_relsymlink` into `helpers.py` and making it "public".

File: commodore/component/compile.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        if config.trace:
+            click.echo(f" > Temp dir left in place {temp_dir}")
+        else:
+            if config.debug:
+                click.echo(f" > Remove temp dir {temp_dir}")
+            shutil.rmtree(temp_dir)
+
+
+def _prepare_fake_inventory(temp_dir: P, component_name, component_path, value_files):
+    component_class_file = component_path / f"class/{component_name}.yml"
```
Comment (simu) [inline commodore/component/compile.py:108]:
```suggestion
    component_class_file = component_path / 'class' / f"{component_name}.yml"
```

File: commodore/component/compile.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    component_defaults_file = component_path / 'class/defaults.yml'
```
Comment (simu) [inline commodore/component/compile.py:109]:
```suggestion
    component_defaults_file = component_path / 'class' / 'defaults.yml'
```

**Comment by srueg** [comment]
I moved as many changes into own PRs as possible. This last one still includes the change of command structure and the actual component compile functionality since they're too closely intertwined.
PR #130 needs to be merged first since it switches to using the Kapitan library instead of running the binary.

File: Makefile [hunk 4]
```diff
@@ -63,10 +65,3 @@ docker:
 		--build-arg GITVERSION=$(GITVERSION) \
 		-t $(IMAGE_NAME) .
 	@echo built image $(IMAGE_NAME)
-
-COMPONENT_TEST_IMAGE_NAME ?= docker.io/vshn/$(BINARY_NAME)-component-test:$(VERSION)
-
-.PHONY: docker-component-test
-docker-component-test:
-	docker build -t $(COMPONENT_TEST_IMAGE_NAME) -f tools/Dockerfile.component_test .
-	@echo built image $(COMPONENT_TEST_IMAGE_NAME)
```
Comment (simu) [inline Makefile:72]: This change doesn't need to be in this PR anymore, this already has been removed in #127

Comment (corvus-ch) [reply Makefile:72]: Sadly we missed that one in #127 so it has to stay.

**Pull request merged**